src/runtime_src/core/tools/xbtracer/src/app/launcher.cpp

```diff
@@ -175,15 +255,82 @@ int parse_cmdline(launcher& app, int argc, char* argv[])
   if (optind == argc)
     log_f("There should be alleast 1 argument without option switch");
 
-  split_by_ws(argv[argc-1], app.m_child_cmd_args);
-  for (auto& element : app.m_child_cmd_args)
+  if (optind <= static_cast<int>(args.size() - 1))
+    app.m_cmdline = args[optind++];
+  else
+    log_f("Invalid optindex: ", optind, "args size: ", args.size());
+
+  for (size_t idx = optind; idx < args.size(); idx++)
   {
-    app.m_cmdline += element;
     app.m_cmdline += " ";
+    app.m_cmdline += args[idx];
   }
-
   log_d("Application to intercept = \"", app.m_cmdline, "\"");
 
+  // Split the last argument using white-space - NOLINTNEXTLINE
+  split_by_ws(argv[argc-1], app.m_child_cmd_args);
+
+  return 0;
+}
+
+int set_envs(launcher& app)
+{
+  /* Adding TRACE_APP_NAME Env */
+  app.m_app_name_env = "TRACE_APP_NAME=";
+  app.m_app_name_env += app.m_cmdline;
+  log_d("Adding to ENV : ", app.m_app_name_env);
+  putenv_t(app.m_app_name_env);
+
+  /* Capture current time */
+  struct timespec start_time = {0, 0};
+  clock_gettime(CLOCK_REALTIME, &start_time);
```

Comment:
It would be better if  `<chrono>` could be used.

Reply:
As suggested, used chrono for reading time. Please check latest update and let me know if any further changes need to be made.